Домашнее задание по 3ему спринту

MatviV2000 opened the pull request:
Сделал сложную задачу по 3ему спринту

MatviV2000 committed:
Выполнено базовое задние:
добавлен label с отображением счетчика и базовым отображением счетчика 0
добавлена кнопка которая увеличивает значение на 1

MatviV2000 committed:
Была выполнена усложненная задача.
Добавил кнопку ресет и -1
Добавил условия работы -1
каждое действие отображается в истории

MatviV2000 committed:
Изменены иконки в соответсвии с заданием

So1ovyov commented on Counter/ViewController.swift:
```diff
@@ -5,15 +5,82 @@
+import Foundation
 import UIKit
 
 class ViewController: UIViewController {
```
Давай добавим модификатор `final` для класса.
Он необходим для оптимизации проекта, также не позволит наследоваться от данного класса.
```swift
final class ViewController: UIViewController {
```

So1ovyov commented on Counter/ViewController.swift:
```diff
@@ -5,15 +5,82 @@
 
 class ViewController: UIViewController {
 
+    @IBOutlet weak var counterLabel: UILabel!
```
Давай поставим модификатор доступа самый закрытый для переменных и функций, как рассказывалось в учебнике, чтобы элементы не были доступны из вне. В том числе к `@IBOutlet` и `@IBAction`. Про это правило помни и в следующих работах тоже.

So1ovyov commented on Counter/ViewController.swift:
```diff
@@ -5,15 +5,82 @@
     override func viewDidLoad() {
         super.viewDidLoad()
-        // Do any additional setup after loading the view.
+        //        incrementButton.tintColor = .systemRed
```
Неиспользуемый код и лишние комментарии принято убирать перед сдачей задачи на ревью. Так мы экономим коллегам время на чтение нашего кода.

So1ovyov commented on Counter/ViewController.swift:
```diff
@@ -5,15 +5,82 @@
+        resetButton.setTitleColor(.systemGray, for: .normal)
     }
+    
+    var counter: Int = 0 {
```
Стоит добавить `private`
Необходимо указывать `private`  для того, чтобы спрятать детали реализации конкретной части функциональности, когда они используются только внутри области объявления.

Здесь и далее в проектах, не забывай проставлять `private` всем свойствам и функциям, которые не планируешь использовать вне класса.

So1ovyov commented on Counter/ViewController.swift:
```diff
@@ -5,15 +5,82 @@
+        )
+    }
+    
+    func scrollUp() {
```
Кажется, что этот метод не нужен, так как в коде не используется, да и в логике проекта не нужно скроллить текст вверх.

MatviV2000 replied:
Здравствуйте!Спасибо за ревью.
Добавил исправления

So1ovyov commented:
@MatviV2000 отличная работа, успехов в следующем спринте!

MatviV2000 committed:
Сделал класс final
Сделал переменные и функции которые работают только в классе private
Убрал не использующийся и закомментированный код

MatviV2000 committed:
Изменена иконка
Блоки кода расставлены в порядке (Иниты, оверрайды, свойства и функции, Экшены (аутлеты пока поставил после оверрайдов. не совсем понял к чему они пренадлежат))
